src/DynamoUtilities/PathHelper.cs

```diff
@@ -31,5 +32,55 @@ public static bool IsValidPath(string folderPath)
         {
             return (!string.IsNullOrEmpty(folderPath) && (File.Exists(folderPath)));
         }
+
+        /// <summary>
+        /// Check if user has readonly privilege to the folder path.
+        /// </summary>
+        /// <param name="folderPath">Folder path</param>
+        /// <returns></returns>
+        public static bool IsReadOnlyPath(string folderPath)
+        {
+            if (IsValidPath(folderPath))
+            {
+                FileInfo Finfo = new FileInfo(folderPath);
```

Comment:
Ah, it's actually a file path, since we already take care the backup we only care if the file is readonly or not. I will update the comment.